index.html

```diff
@@ -24,125 +41,629 @@
+            data-btn
+            data-popover-control
+          >
+            Toggle Popover with Layered Styles
+          </button>
+          <pre><code class="language-html"
+>&lt;button popovertarget="layeredStylesPopover"&gt;
+  Toggle Popover with Layered Styles
+&lt;/button&gt;
+
+&lt;div id="layeredStylesPopover" data-popover="layered-styles" popover&gt;
+  Popover with Layered Styles
+  (should be red if browser has &lt;code&gt;@layer&lt;/code&gt; support)
+&lt;/div&gt;</code>
+
+<code class="language-css"
+>/* CSS */
+@layer popover-polyfill;
+
+@layer consumer {
+  [data-popover~='layered-styles'] {
+    background-color: #d00d1e;
+  }
+}</code></pre>
+        </section>
+        <section id="popover-shadow-root" class="demo-item">
+          <h2 data-header>
+            <a href="#popover-shadow-root" aria-hidden="true">🔗</a>
+            Popovers and Controls Created Fully in the Shadow Root
+          </h2>
+          <p class="note">
+            Here, both the popover control button and the popover are created in
+            the Shadow DOM.
+          </p>
+          <div id="fullShadowHost"></div>
+          <pre><code class="language-html"
+>&lt;!-- Shadow DOM --&gt;
+&lt;button popovertarget="shadowedPopover"&gt;
+  Toggle Shadowed Popover
+&lt;/button&gt;
+&lt;button popovertarget="shadowedNestedPopover"&gt;
+  Toggle Shadowed Nested Popover
+&lt;/button&gt;
+
+&lt;div id="shadowedPopover" popover&gt;Shadowed Popover&lt;/div&gt;
+&lt;div&gt;
+  &lt;div id="shadowedNestedPopover" popover&gt;Shadowed Nested Popover&lt;/div&gt;
+&lt;/div&gt;</code></pre>
+        </section>
+        <script type="module">
+          const fullShadowHost = document.getElementById('fullShadowHost');
+          const fullShadowRoot = fullShadowHost.attachShadow({ mode: 'open' });
+          fullShadowRoot.innerHTML = `
+            <style>
+              @layer reset, initial, popover-polyfill, default, demo;
+
+              @import url('/css/reset.css') layer(reset);
+              @import url('/css/demo.css') layer(demo);
+              @import url('/css/prism.css') layer(initial);
+              @import url('/css/fonts.css') layer(initial);
+              @import url('/css/base.css') layer(default);
+              @import url('/css/props.css') layer(initial);
+              @import url('/css/code.css') layer(default);
+            </style>
+            <div class="button-group" data-popover-control>
+              <button popovertarget="shadowedPopover" data-btn>
+                Toggle Shadowed Popover
+              </button>
+              <button popovertarget="shadowedNestedPopover" data-btn>
+                Toggle Shadowed Nested Popover
+              </button>
+            </div>
+            <div id="shadowedPopover" popover data-popover>Shadowed Popover</div>
+            <div>
+              <div id="shadowedNestedPopover" popover data-popover>Shadowed Nested Popover</div>
+            </div>`;
```

Comment:
For some reason these popovers are centered vertically in the document, not in the viewport, so you have to scroll way up to see that they're open. Any idea what's causing that?

Reply:
See https://github.com/oddbird/popover-polyfill/pull/187#discussion_r1583652526